hardware_interface/include/hardware_interface/component_interfaces/actuator_interface.hpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__COMPONENT_INTERFACES__ACTUATOR_INTERFACE_HPP_
+#define HARDWARE_INTERFACE__COMPONENT_INTERFACES__ACTUATOR_INTERFACE_HPP_
+
+#include <algorithm>
+#include <array>
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
```

Comment:
I agree to use that. Is it feasible to expand this with additional codes? Something similar I did for `components_ret_t` in robot_control_components/ros2_control_types.h

I intend to inform upper-level components on type of error is happening and hopefully mitigate it, or at least provide better error output for users.

Reply:
absolutely. This is basically inspired from here: https://github.com/ros2/rcl/blob/master/rcl/include/rcl/types.h